spec/tiler_spec.coffee

```diff
@@ -1,92 +1,198 @@
+
+    context 'with custom activeTile', ->
+      before ->
+        $tiler = $('#active-tile')
+        $tiler.tiler
+          activeTile: 2
 
-    describe '_buildLinks', ->
-      it 'should add tile data to the link', ->
-        expect($('button').data('tiler-title')).to.equal 'Tile 1'
-        expect($('button').data('tiler-foo')).to.equal 'Foo 1'
+      it 'should set the active & pervious tiles', ->
+        expect($('#tile-active-tile-2', $tiler)).to.have.class 'active'
+        expect($('#tile-active-tile-2', $tiler)).to.not.have.class 'previous'
+        expect($('#tile-active-tile-1', $tiler)).to.have.class 'previous'
+        expect($('#tile-active-tile-1', $tiler)).to.not.have.class 'active'
 
-    context 'with options', ->
-      describe 'isReversible', ->
-        before ->
-          $('#reverse-support').tiler
-            isReversible: true
-          $('#reverse-support').tiler('goTo', 2)
-          $('#reverse-support').tiler('goTo', 1)
+  describe 'goTo', ->
+    before ->
+      $tiler = $('#go-to')
+      $tiler.tiler()
+      $tiler.tiler 'goTo', 2, false
 
-        it 'should set the reverse classes', ->
-          expect($('#reverse-support #tile-1').hasClass('reverse')).to.be.true
+    it 'should set the active tile id on the viewport', ->
+      expect($tiler.attr('data-tiler-active-tile')).to.equal 'tile-go-to-2'
 
-    describe '_setupTiles', ->
-      context 'when viewport has a height', ->
-        before ->
-          $('#viewport-height').tiler()
+    context 'when going to the current tile', ->
+      currentTileViewportSpy = null
 
-        it 'should match the tiles size to the viewport size', ->
-          expect($('#viewport-height .tiler-tile').outerWidth()).to.equal 321
-          expect($('#viewport-height .tiler-tile').outerHeight()).to.equal 123
+      before ->
+        currentTileViewportSpy = sinon.spy()
+        $tiler = $('#current-tile')
+        $tiler.one 'tiler.goto', (e, data) ->
+          currentTileViewportSpy data
+
+        $tiler.tiler()
+        $tiler.tiler 'goTo', 1, false
+
+      it 'should not fire events', ->
+        expect(currentTileViewportSpy).to.not.be.called
+
+      it 'should not update the tile classes', ->
+        # tile 1
+        expect($('#tile-current-tile-1', $tiler)).to.have.class 'active'
+        expect($('#tile-current-tile-1', $tiler)).to.have.class 'enter'
+        expect($('#tile-current-tile-1', $tiler)).to.not.have.class 'previous'
+        expect($('#tile-current-tile-1', $tiler)).to.not.have.class 'exit'
+
+        # tile 2
+        expect($('#tile-current-tile-2', $tiler)).to.have.class 'previous'
+        expect($('#tile-current-tile-2', $tiler)).to.have.class 'exit'
+        expect($('#tile-current-tile-2', $tiler)).to.not.have.class 'active'
+        expect($('#tile-current-tile-2', $tiler)).to.not.have.class 'enter'
+
+    context 'with events', ->
+      viewportEventSpy = null
```

Comment:
so you consider these block scoped vars? we don't need to declare them at the top of the file?